docs/cloud-edge/modules/user-agent-filter.md

```diff
@@ -0,0 +1,130 @@
+# User Agent Filter
+
+---
+
+## Overview
+
+With the User Agent Filter module you can allow or deny traffic through regular expression-based rules that are matched against the incoming user-agent header, which contains information about the requesting browser or application, including its name, version, and operating system.
+
+Since rules are processed at the ngrok Cloud Edge, you can bolster your application’s security by blocking unwanted or unauthorized requests from known bots and malicious actors before it ever reaches your network.
+
+## Quickstart
+
+### Agent CLI
+
+```bash
+ngrok http 80 \
+  --ua-filter-allow="(foo/(\d)+.(\d)+)","(bar/(\d)+.(\d)+)" \
+  --ua-filter-deny="(buz/(\d)+.(\d)+)"
+```
+
+### Agent Configuration File
+
+```yaml
+tunnels:
+  example:
+    proto: http
+    addr: 80
+    ua_filter_allow: ["(foo/(\d)+.(\d)+)", "(bar/(\d)+.(\d)+)"]
+    ua_filter_deny: ["(buzz/(\d)+.(\d)+)"]
+```
+
+### Go SDK
+
+```go
+import (
+	"context"
+	"net"
+
+	"golang.ngrok.com/ngrok"
+	"golang.ngrok.com/ngrok/config"
+)
+
+func listenUserAgentFIlter(ctx context.Context) net.Listener {
+	listener, _ := ngrok.Listen(ctx,
+		config.HTTPEndpoint(
+			config.WithUserAgentFilter([]string{`(foo)/(\d)+.(\d)+`}, []string{`bar/(\d)+.(\d)+`}),
+		),
+		ngrok.WithAuthtokenFromEnv(),
+	)
+	return listener
+}
+```
+
+## Behavior
+
+### Rule Evaluation
+
+On request, the User Agent Filter module will check the incoming `User-Agent` header value against the list of defined allow and deny regular expression rules.
+
+- Requests without the `User-Agent` header will be denied by default. To allow these requests, you can add an `allow` rule for empty strings: `^$`
+- Requests with multiple `User-Agent` headers will only match the first header.
+
+### Match Behavior
+
+Requests are allowed if they match an `allow` rule or no matching `deny` rules. Requests that match `deny` rules are rejected with a `403 Forbidden` error.
```

Comment:
> Requests are allowed if they match an `allow` rule or no matching `deny` rules.

This sentence doesn't quite make sense. Do we want to say:

Requests are allowed if they match any `allow` rule or fail to match all `deny` rules.